arbor/backends/gpu/matrix_state_fine.hpp

```diff
@@ -83,52 +83,60 @@ struct matrix_state_fine {
     using const_view = typename array::const_view_type;
     using iarray     = memory::device_vector<size_type>;
 
-    template <typename ValueType>
-    using managed_vector = std::vector<ValueType, memory::managed_allocator<ValueType>>;
+    using metadata_array = memory::device_vector<level_metadata>;
 
     iarray cv_to_cell;
 
     array d;     // [μS]
     array u;     // [μS]
     array rhs;   // [nA]
 
-    // required for matrix assembly
-
-    array cv_area; // [μm^2]
-
+    // Required for matrix assembly
+    array cv_area;             // [μm^2]
     array cv_capacitance;      // [pF]
 
-    // the invariant part of the matrix diagonal
+    // Invariant part of the matrix diagonal
     array invariant_d;         // [μS]
 
-    // for storing the solution in unpacked format
+    // Solution in unpacked format
     array solution_;
 
+    // Maps cell to integration domain
     iarray cell_to_intdom;
 
-    // the maximum nuber of branches in each level per block
+    // Maximum number of branches in each level per block
     unsigned max_branches_per_level;
 
-    // number of rows in matrix
+    // Number of rows in matrix
     unsigned matrix_size;
 
-    // number of cells
+    // Number of cells
     unsigned num_cells;
-    managed_vector<unsigned> num_cells_in_block;
+    iarray num_cells_in_block;
 
-    // end of the data of each level
-    managed_vector<unsigned> data_partition;
+    // End of the data of each level
+    iarray data_partition;
     std::size_t data_size;
 
-    // the meta data for each level for each block layed out linearly in memory
-    managed_vector<level> levels;
-    // the start of the levels of each block
-    // block b owns { leves[level_start[b]], ..., leves[level_start[b+1] - 1] }
+    // Metadata for each level
+    // Includes indices into d, u, rhs and levels_lengths, levels_parents
+    metadata_array levels_meta;
```

Comment:
The plural on `levels` is a bit confusing, I would rename these to `level_meta`, `level_lengths` etc.

Reply:
Done.